saga/sagaState.go

```diff
@@ -6,30 +6,40 @@ import (
 )
 
 /*
+ * Additional data about tasks to be committed to the log
+ * This is Opaque to SagaState, but useful data to persist for
+ * results or debugging
+ */
+type taskData struct {
```

Comment:
This [pull requests] (https://github.com/scootdev/scoot/pull/14) reduces most of them.  So with both merged there will be 
* map of taskId -> flag
* map of taskId -> taskData

In theory yes we could combine the two next if we want which will save an extra string copy per task.    

I don't understand the immutable comment?  Go doesn't provide Immutable constructs.  This code provides SagaState Immutability based on the API (only read methods outside of package) and based on the copyMethod, which is updated to treat taskData as immutable by copying the (ptr, len, cap) slice object when copying a SagaState. 

There's only so much I can do to prevent a user from mutating the underlying []bytes since we give them the slice reference back with a pointer to the underlying slice.